source/pcic/client.py

```diff
@@ -36,6 +36,7 @@ def connect(self):
         if not self.connected:
             self.pcicSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             self.pcicSocket.connect((self.address, self.port))
+            self.pcicSocket.settimeout(self.timeout)
```

Comment:
@bnozka Thank you for the review. I've didn't got the mail. 

1. 2. Sure no problem.
3. I would like to not wait for timeouts or connect to anything or use sockets at all with unit tests. We can mock sockets and throw timeouts from mock. What do you think about that approach? Btw. We will need to change/remake unit tests for stuff like this and using pyTest. It would be perfect to run them on github per each commit/push request.
4. I am unsure what `socket_exception_handler` does. Can we solve this problem in another PR?
5. What would be the pros of that?

Reply:
What is the difference between `disconnect` and `close`?

Reply:
@bnozka 

1. ok
2. ok

3. Looks good to me. Changing all unittests to pytests is a huge change for all tests. ~approx. 100 tests at all. @cfreundl 
4. `socket_exception_handler `has the same behaviour like `settimeout`, but was only used for the connecting step. There was no impact on further  methods. But yes, we can delete this in another PR.

5. One of our customers implemented own `IfmSocketConnectTimeout `and `IfmSocketRecvTimeout ` exceptions for faster and better understanding what was wrong. Another reasons is that if he wants to handle the socket timeout exception he needs to import it like this:

```
from source.pcic.client import O2x5xxPCICDevice

from socket import timeout as socket_timeout

try:
    device = O2x5xxPCICDevice("192.168.0.55", 50010)
except socket_timeout as e:
    print(e)
```

6. There is no difference. The disconnect is used by a customer. The close is frequently used internally. Is there a handy way to fix that?